### PR TITLE
Partitioning

### PR DESCRIPTION
**Pull request by mdhorn**
Fixes Issue: #14 

Changes proposed in this pull request:
- New TUI interface for media selection
- Backend changes to ensure we do not overwrite existing media during Safe Install
- Media will be overwritten if Destructive Install is chosen by the user
- Launch disk partitioning tool for selected media to allow the user to create free space for the install
  - Note: Resizing a partition does NOT resize a file system by default

### CODE REVIEW

File: model/model.go [hunk 1]
```diff
@@ -41,6 +41,12 @@ var testAlias = []string{}
 // SystemInstall represents the system install "configuration", the target
 // medias, bundles to install and whatever state a install may require
 type SystemInstall struct {
+	// Even though TargetMedias is an array, we currently only support
+	// it containing a single BlockDevice which matches/matps to the InstallTarget
```
Comment (reaganlo) [inline model/model.go]: Spell check  "matches/maps"

Comment (mdhorn) [reply model/model.go]: fixed

File: tui/disk_util.go [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+	contentFrame.SetPack(clui.Vertical)
+
+	page.descrptionLabel = clui.CreateLabel(contentFrame, AutoSize, 7, "", Fixed)
+	page.descrptionLabel.SetMultiline(true)
+	page.descrptionLabel.SetPaddings(0, 2)
+	descriptiveMessage := "Launch an external program (" + diskUtil +
+		") to modify the selected\n" +
+		"disk in an attempt to allocate enough free disk space to allow\n" +
+		"the installation to succeed.\n\n" +
+		"Only disk with a 'gpt' partition can be modifify.\n\n" +
```
Comment (reaganlo) [inline tui/disk_util.go]: Spellcheck "modify" instead of "modifify"

File: tui/media_config.go [hunk 3]
```diff
@@ -0,0 +1,427 @@
...
+					warning = fmt.Sprintf("Warning: %s", warning)
+					page.labelWarning.SetTitle(warning)
+				}
+			}
+		}
+	})
+
+	// Description of Safe
+	descLabel := clui.CreateLabel(radioButtonFrame, 2, 3, "Uses non-partitioned or gpt partitioned media with free space.\n"+
+		"Install along side existing data or OS if adequate space is available.", Fixed)
```
Comment (reaganlo) [inline tui/media_config.go]: Usage. Change "along side" to one word "alongside"

File: tui/disk_util.go (hunk 2 above)
Comment (mdhorn) [reply tui/disk_util.go]: fixed

File: tui/media_config.go (hunk 3 above)
Comment (mdhorn) [reply tui/media_config.go]: fixed in new force push

**Comment by reaganlo** [comment]
Few things Mark:
- Is it intentional to have the media list change only on "**double click**" of the Installation Type instead of "**single click**"?
-  **Auto partition** gives a warning and a **Dismiss** button. Not able to proceed further with the Auto partition. Is that intentional?
- During manual partition, when I select each partition and reduce the size (e.g by a few MB's), it gives the message size too large.

**Comment by mdhorn** [comment]
> @mdhorn Few things Mark:
> 
> * Is it intentional to have the media list change only on "**double click**" of the Installation Type instead of "**single click**"?
> * **Auto partition** gives a warning and a **Dismiss** button. Not able to proceed further with the Auto partition. Is that intentional?
> * During manual partition, when I select each partition and reduce the size (e.g by a few MB's), it gives the message size too large.

I can't seem to find the "double click" issue -- I'll stop by.  But in general, the TUI does not officially support mouse, only keyboard.

The Auto Partition warns because you are changing a disk with content. After it is dismissed, the disk you selected should have already been partitioned and you are ready to hit the Confirm button. If that's not the case, I need to see your test case.

The "size too large" is a bug in that we switched everything in manual partitioning from MiB to MB, but the check on the size was still checking MiB. I'm pushing a fix for this.

**Pull request merged**